casper/src/main/scala/coop/rchain/casper/util/rholang/SystemDeployUtil.scala

```diff
@@ -0,0 +1,58 @@
+package coop.rchain.casper.util.rholang
+
+import com.google.protobuf.{ByteString}
+import java.io.{ByteArrayInputStream, ByteArrayOutputStream}
+
+import com.google.protobuf.{CodedInputStream, CodedOutputStream}
+import com.google.protobuf.wrappers.Int32Value
+import coop.rchain.casper.genesis.contracts.Validator
+import coop.rchain.casper.protocol.DeployData
+import coop.rchain.crypto.hash.Blake2b512Random
+import coop.rchain.crypto.signatures.Signed
+import coop.rchain.crypto.PublicKey
+
+object SystemDeployUtil {
+  import coop.rchain.models.Validator.Validator
+  // Currently we have 4 system deploys -> refund, preCharge, closeBlock, Slashing
+  // In every user deploy, the rnode would do the preCharge first, then execute the
+  // user deploy and do the refund at last.
+  //
+  // The refund and preCharge system deploy
+  // would use user deploy signature to generate the system deploy. The random seed of
+  // the refund and preCharge has to be exactly the same to make sure replay the user
+  // deploy would come out the exact same result.
+  //
+  // As for closeBlock and slashing, the rnode would execute closeBlock system deploy in every block.
+  // So for a block seed it would be enough to have:
+  // PREFIX ++ PublicKey ++ seqNum serialized with protobuf.
+  // This way we can be completely sure that collision cannot happen.
+  // (Quote: https://github.com/rchain/rchain/pull/2879#discussion_r378948921)
+
+  private def serializeInt32Fixed(value: Int): Array[Byte] = {
+    val stream = new ByteArrayOutputStream
+    val proto  = CodedOutputStream.newInstance(stream)
+    proto.writeFixed32NoTag(value)
+    proto.flush()
+    stream.toByteArray
+  }
+  def generateSystemDeployRandomSeed(sender: Validator, seqNum: Int): Blake2b512Random =
+    Tools.rng(serializeInt32Fixed(0) ++ sender.toByteArray() ++ serializeInt32Fixed(seqNum))
```

Comment:
Please extract as a variable this prefix, something like `SYSTEM_DEPLOY_PREFIX`. We want to communicate that this is important info.
I was thinking about serialized string as a prefix, but I like your choice better.